client/scala/src/main/scala/ai/verta/blobs/dataset/Dataset.scala

```diff
@@ -38,4 +42,13 @@ trait Dataset extends Blob {
 
   /** Get the set of all the files' metadata managed by the Dataset blob  */
   def getAllMetadata = contents.values
+
+  /** Check if the other dataset is combinable (i.e no conflicting entries)
+   *  @param other other dataset to combine
+   *  @return whether there is a conflict in the contents of two dataset
+   */
+  protected def canCombine(other: Dataset) = {
```

Comment:
`conflicts` would be a better name here from the description, right?

Reply:
I swear I name it conflicts initially, then change it to canCombine =) Ok I'll fix.

Reply:
Fixed.